multierror.go

```diff
@@ -49,3 +50,7 @@ func (e *Error) GoString() string {
 func (e *Error) WrappedErrors() []error {
 	return e.Errors
 }
+
+func (e *Error) MarshalJSON() ([]byte, error) {
+	return json.Marshal(e.Errors)
```

Comment:
As we discussed in person the following would cover both marshal and unmarshal:
```go
 type Error struct {
 	Errors      []error         `json:"errors"`
 	ErrorFormat ErrorFormatFunc `json:"-"`
 }
```
but since the majority of errors aren't marshalable, do we want to do it in the library at all?